python/ray/_private/runtime_env/validation.py

```diff
@@ -103,17 +109,56 @@ def parse_and_validate_conda(conda: Union[str, dict]) -> Union[str, dict]:
     return result
 
 
+def _rewrite_pip_list_ray_libraries(pip_list: List[str]) -> List[str]:
+    result = []
+    for specifier in pip_list:
+        requirement = Requirement.parse(specifier)
+        package_name = requirement.name
+        if package_name == "ray":
+            libraries = requirement.extras  # e.g. ("serve", "tune")
+            if libraries == ():
+                # Ray alone was specified (e.g. "ray" or "ray>1.4"). Remove it.
+                if os.environ.get(RAY_RUNTIME_ENV_ALLOW_RAY_IN_PIP) != "1":
+                    logger.warning(
+                        "Ray was specified in the `pip` field of the "
+                        f"`runtime_env`: '{specifier}'. This may cause version"
+                        " compatibility issues, so it has been automatically "
+                        "deleted from the `pip` field, and the Ray version "
+                        "already installed on the cluster will be used.  To "
+                        "disable this behavior, set the environment variable "
+                        f"{RAY_RUNTIME_ENV_ALLOW_RAY_IN_PIP} to 1.")
```

Comment:
In most cases users will have passed this through accidentally, mention that they don't need to do it in the `pip` field?